src/dao/user.dao.ts

```diff
@@ -28,6 +28,25 @@ class UserDAO {
     });
   }
 
+async fetchUsers(
+  filter: { username?: string; email?: string }, 
+  page: number = 1, 
+  pageSize: number = 10
+) {
+  const skip = (page - 1) * pageSize;
+  const where = {};
+
+  if (filter.email) where.email = { contains: filter.email };
```

Comment:
Hey folks! nice to see this moving forward!

Question here, when creating the where filter this way, would both where clauses be respected? This would be a AND or OR where?

I believe that prisma handles this differently. Remember, in this query we want an OR condition for the where, please check this reference! https://www.prisma.io/docs/orm/prisma-client/queries/filtering-and-sorting#combining-operators